backend/src/main/java/com/redhat/cloud/notifications/oapi/OApiFilter.java

```diff
@@ -0,0 +1,203 @@
+                .put("version", "1.0")
+                .put("title", uppify(what)));
+
+        // Add servers section
+        JsonArray serversArray = new JsonArray();
+
+        if (what.equals(NOTIFICATIONS)) {
+            serversArray.add(createServer(true, NOTIFICATIONS));
+            serversArray.add(createServer(false, NOTIFICATIONS));
+        } else if (what.equals(INTEGRATIONS)) {
+            serversArray.add(createServer(true, INTEGRATIONS));
+            serversArray.add(createServer(false, INTEGRATIONS));
+        }
+
+        root.put("servers", serversArray);
+
+        return root;
+    }
+
+    private String uppify(String what) {
+        return what.substring(0, 1).toUpperCase() + what.substring(1);
+    }
+
+    private JsonObject filterPrivateOperation(JsonObject pathObject, boolean removePrivate) {
+        JsonObject newPathObject = new JsonObject();
+        pathObject.stream().forEach(entry -> {
+            String verb = entry.getKey();
+            JsonObject operation = (JsonObject) entry.getValue();
+            boolean isPrivate = operation.containsKey("tags") && operation.getJsonArray("tags").contains(PRIVATE);
+            if (isPrivate != removePrivate) {
+                newPathObject.put(verb, operation);
+            }
+        });
+
+        if (newPathObject.size() == 0) {
+            return null;
+        }
+
+        return newPathObject;
+    }
+
+    private JsonObject createServer(boolean isProd, String what) {
+
+        JsonObject job = new JsonObject();
+        if (isProd) {
+            job.put("url", "https://cloud.redhat.com");
+            job.put("description", "Production Server");
+            job.put("variables", new JsonObject()
+                    .put("basePath", new JsonObject()
+                            .put("default", "/api/" + what + "/v1.0")));
+        } else {
+            job.put("url", "http://localhost:{port}");
+            job.put("description", "Development Server");
+            job.put("variables", new JsonObject()
+                    .put("basePath", new JsonObject()
+                            .put("default", "/api/" + what + "/v1.0"))
+                    .put("port", new JsonObject()
+                            .put("default", "8080")));
+        }
+        return job;
+    }
+
+    private String mangle(String in) {
+        String out = null;
+        if (in.startsWith(Constants.API_INTEGRATIONS_V_1_0)) {
+            out = in.substring(Constants.API_INTEGRATIONS_V_1_0.length());
+        }
+        if (in.startsWith(Constants.API_NOTIFICATIONS_V_1_0)) {
+            out = in.substring(Constants.API_NOTIFICATIONS_V_1_0.length());
+        }
+        if (in.startsWith(Constants.INTERNAL)) {
+            out = in.substring(Constants.INTERNAL.length());
+        }
+        if (out != null && out.isEmpty()) {
+            out = "/";
+        }
+
+        if (out == null) {
+            System.out.println("Out is null and in is:" + in);
```

Comment:
Should we use a logger here?

Reply:
Actually this was something quick to debug :D (Removing)